p2p/trust/trustmetric.go

```diff
@@ -563,13 +561,17 @@ loop:
 				tm.good = 0
 				tm.bad = 0
 			}
+			tm.mtx.Unlock()
 		case stop := <-tm.stop:
+			tm.mtx.Lock()
```

Comment:
we don't need the lock until we want to set pause. so we can check `stop`, and if it's false, call `tm.Pause()` which should handle the locking.